src/modules/dyad.c

```diff
@@ -126,15 +127,41 @@ dyad_fetch_request_cb (flux_t *h,
     char *upath = NULL;
     char fullpath[PATH_MAX + 1] = {'\0'};
     int saved_errno = errno;
-    errno = 0;
+    dyad_rc_t rc = 0;
 
-    if (flux_msg_get_userid (msg, &userid) < 0)
-        goto error;
+    if (!flux_msg_is_streaming (msg)) {
```

Comment:
We should have a discussion on what the best terms is for "streaming" later.

Reply:
I agree. FWIW, in this context, the term "streaming" refers to Flux's streaming RPCs (i.e., an RPC which can send multiple return messages and is ended with an error return message with errno set to `ENODATA`).